docs/source/faq.rst

```diff
@@ -110,3 +106,43 @@ For the Arrow ecosystem, we hope databases will implement the Flight
 SQL wire protocol, giving them access to all the Flight SQL clients,
 including ADBC, JDBC, and ODBC drivers.  (And ADBC users can still use
 other drivers to work with other databases.)
+
+So what is the "ADBC Flight SQL driver" then?
+=============================================
+
+The ADBC Flight SQL driver implements the ADBC API standard (which an
+application interacts with) using the Flight SQL wire protocol (which
+a database server exposes).  So it's a generic driver that can talk to
+many databases, and it also implements a generic API (ADBC) designed
+to abstract over multiple databases.
```

Comment:
Is the "and it also implements a generic API (ADBC) ..)" needed? Because that is what an ADBC driver is in general, no? (something that implements the generic ADBC API)

Another suggestion, could also do "So it's a generic driver that can talk to many databases, as long as those implement Flight SQL".